packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -549,11 +555,113 @@ export class NetworkMonitor {
     return deployments
   }
 
+  async latestValidEpoch(network?: string): Promise<NetworkEpochBlock> {
+    network = network ?? this.blockOracleSubgraph.network
+    this.logger.trace(`latest valid epoch`, {
+      settlementNetwork: this.blockOracleSubgraph.network,
+      network,
+      networkAlias: NetworkAliases[network],
+    })
+    try {
+      const result = await this.blockOracleSubgraph.query(
+        gql`
+          query globalStates($network: String!) {
+            globalStates {
+              latestValidEpoch {
+                epochNumber
+                blockNumbers(where: { network: $network }) {
+                  network {
+                    id
+                  }
+                  blockNumber
+                  previousBlockNumber {
+                    blockNumber
+                  }
+                }
+              }
+            }
+          }
+        `,
+        {
+          network: NetworkAliases[network],
+        },
+      )
+
+      if (result.error) {
+        throw result.error
+      }
+
+      if (!result.data.globalStates) {
+        throw indexerError(
+          IndexerErrorCode.IE069,
+          `Failed to query latest valid epoch number for the global state`,
+        )
+      }
+
+      const epochBlock = await buildEpochBlock(
+        this.indexingStatusResolver,
+        result.data.globalStates[0].latestValidEpoch.epochNumber,
```

Comment:
Just a heads up:
We will change the schema a little bit once https://github.com/graphprotocol/graph-node/pull/3096 is merged. The `epochNumber` field will most likely be exclusive to the `Epoch` entity. 

The change won't affect this query, but it might introduce a more convenient way to do it. I can help with this update when that happens.

Reply:
Yes, I updated it such that we query only one `globalState` with id specified to 0, so we don't need to use indices